src/main.rs

```diff
@@ -66,12 +65,18 @@ async fn main() -> Result<(), LemmyError> {
   );
 
   let activity_queue = create_activity_queue();
+
+  // Required because docker installs try to use TLS, which is disabled as a reqwest feature flag
+  let client = reqwest::ClientBuilder::new()
+    .danger_accept_invalid_certs(true)
+    .build()?;
```

Comment:
We should only use this for testing, not in prod.

Reply:
This fails for both dev and prod. The only way to not do this would be to turn on an SSL feature flag of reqwest.

Reply:
[And thats enabled by default](https://github.com/seanmonstar/reqwest/blob/master/Cargo.toml#L29). Its also required by activitypub, so its impossible to disable it. Not to mention that it would be a security vulnerability for open graph fetching as well. 

So I really dont understand why you think this is necessary.

Reply:
It took me a while to find the ssl bug, but the docker post creates will fail without this flag. I'll redo it then post the error.

Reply:
Yep, just tested removing it, and it gets this error every time: 

`error sending request for url (https://www.marxists.org/archive/lenin/works/1921/apr/21.htm): error trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:../ssl/statem/statem_clnt.c:1913: (unable to get local issuer certificate)`

Reply:
Just realized this is the same error as https://github.com/LemmyNet/lemmy/issues/1444

Reply:
Yes something is wrong with ssl in our dev docker images, so we could use this workaround only for dev builds. Or better yet try to fix it somehow. But we definitely cant disable cert checks for prod.

Reply:
I checked the dockerfiles for prod and dev, there's no additional SSL related differences, why would one image work but not the other?

Reply:
The other possibility is to switch to rusttls from native-tls, but you switched away from this for build time reasons, right?

Reply:
Iirc we were compiling rustls for no reason, and were using native tls anyway (which we are still using now). 

I have no idea why it doesnt work in Docker. There are some results about this error online, [maybe this fix would work](https://serverfault.com/questions/449599/ssl-certificate-unable-to-get-local-issuer-certificate).

Reply:
I tried compiling reqwest with `rustls-tls`

`reqwest = { version = "0.11.4", default-features = false, features = ["json", "rustls-tls"] }`

And it gave the same error. 

I've also tried using `openssl_probe::init_ssl_cert_env_vars();` as recommended [here.](https://hub.docker.com/r/ekidd/rust-musl-builder). Neither worked.

Reply:
I would open a separate issue, as it has nothing to do with this PR.

Reply:
If we merge this without using `.danger_accept_invalid_certs(true)`, all post creates will be broken.

Reply:
But the SSL problem doesnt happen in prod, otherwise federation wouldnt work either.

Reply:
Ohhh okay I can maybe guess the issue. We never use or generate self-signed certs (which apparently reqwest needs ) , we use letsencrypt for the ansible deploys, which docker is somehow smart enough to be able to insert into from the system into our docker container, which rust can then use. I have no idea how it does this without any configuring, because we haven't touched a thing or passed these forward in any way.

The best way would be to check if it existsusing openssl_probe: https://docs.rs/openssl-probe/0.1.4/openssl_probe/fn.has_ssl_cert_env_vars.html , and use the dangerous flag if there are no certs, and output a warning in the console.

Reply:
Reqwest as http client doesnt need access to website certificates, those are sent by the server. All it needs are the CA certificates in order to verify the tls certificate. Did you try `apt-get install ca-certificates` and checking `CURL_CA_BUNDLE` env var as described in my link above?

Reply:
Ah that appeared to do it, I forgot that the `volume_mount` uses a different DockerFile.

I wonder why this doesn't need to be done with the musl images.